qiskit/chemistry/qubit_transformations/qubit_operator_transformation.py

```diff
@@ -0,0 +1,36 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2018, 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Base class for transformation to qubit operators for chemistry problems"""
+
+from abc import ABC, abstractmethod
+from typing import Tuple, List
+
+from qiskit.aqua.operators.legacy import WeightedPauliOperator
+from qiskit.chemistry.core import MolecularGroundStateResult
+from qiskit.chemistry.drivers import BaseDriver
+
+
+class QubitOperatorTransformation(ABC):
+    """Base class for transformation to qubit operators for chemistry problems"""
+
+    @abstractmethod
+    def transform(self, driver: BaseDriver
```

Comment:
You address multiple things here:

* use of `WeightedPauliOp`: I tend to agree, but the transformations will need to undergo another refactoring in the future, anyways, so I am willing to except this temporary solution.
* naming: will be discussed towards the end of this PR.
* docstring: will also be addressed at the end.
* `BaseDriver`: this should be resolved by #1297

Reply:
>use of WeightedPauliOp: I tend to agree, but the transformations will need to undergo another refactoring in the future, anyways, so I am willing to except this temporary solution.

The intent is to deprecate these operators - which is why they are in legacy. Building out new function with these seems rather suspect in my mind. Yes internally we will have to continue to use them but as soon as opflow has Z2Symmetry support that would no longer need to be the case. Yes there is the future of opflow to consider too. Since it is so easy given a WeightedPauliOperator to call the to_opflow() to get an opflow equivalent I would think we would do this here so the new interface emits new types and not legacy. When passed to VQE, which for legacy reasons still accepts these too, it first converts to opflow anyway. I am far less willing to accept new APIs being built out with these given it seems such an easy change.

Reply:
to be pushed soon